airflow/ui/src/pages/DagsList/DagsList.tsx

```diff
@@ -89,35 +91,57 @@ const columns: Array<ColumnDef<DAGResponse>> = [
   },
 ];
 
+const { PAUSED: PAUSED_PARAM, NAME_PATTERN: NAME_PATTERN_PARAM } = searchParamsKeys;
+
+// eslint-disable-next-line complexity
 const cardDef: CardDef<DAGResponse> = {
   card: ({ row }) => <DagCard dag={row} />,
   meta: {
     customSkeleton: <Skeleton height="120px" width="100%" />,
   },
 };
 
-const PAUSED_PARAM = "paused";
-
 export const DagsList = () => {
-  const [searchParams] = useSearchParams();
+  const [searchParams, setSearchParams] = useSearchParams();
   const [display, setDisplay] = useState<"card" | "table">("card");
 
   const showPaused = searchParams.get(PAUSED_PARAM);
 
   const { setTableURLState, tableURLState } = useTableURLState();
   const { pagination, sorting } = tableURLState;
 
+  const searchBarRef = useRef<HTMLInputElement>(null);
+
   // TODO: update API to accept multiple orderBy params
   const [sort] = sorting;
   const orderBy = sort ? `${sort.desc ? "-" : ""}${sort.id}` : undefined;
 
+  const [dagDisplayNamePattern, setDagDisplayNamePattern] = useState(searchParams.get(NAME_PATTERN_PARAM) ?? undefined);
+
+  const dagDisplayNamePatternDebounceDelay = 200;
+
+  const handleSearchBarChange = useDebouncedCallback(
+    ({ target }: SyntheticEvent<HTMLInputElement>) => {
+      const { value } = target as HTMLInputElement;
```

Comment:
```suggestion
    ({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
```

This should remove the need for `as`